### PR TITLE
DecalRendererFeature: Convert to using RTHandles

### PR DESCRIPTION
**Pull request by sandy-carter-unity**
Pass: Remove RTI alises of universal renderer targets, these do not work
with RTHandles, you must get the actual references.
Pass: Convert dBufferColorHandles[0-2] and m_DBufferDepth to RTHandles
Pass: Set global textures in execute to use the same cmd.
Feature: Call pass dispose
Feature: Move setup of targets to Pass' Setup Function called from
SetupRenderPasses

# **Please read the [Contributing guide](CONTRIBUTING.md) before making a PR.**

* Read the [Graphics repository & Yamato FAQ](http://go/graphics-yamato-faq).

### Checklist for PR maker
- [ ] Have you updated the changelog? Each package has a `CHANGELOG.md` file.

---
### Purpose of this PR
The intention of this PR is fix breaking test 230 in Foundation.
It also fixes targets that weren't yet converted to RTHandles.

---
### Testing status
Test 230_Decal_Projector from UniversalGraphicsTest_Foundation was run with the `-xr-reuse-tests` which reliably produced a red test.

The test seemed to also fail more often if run after 120 to 124. This was also tested.

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

**Comment by sandy-carter-unity** [comment]
The following red nightly tests tests are fixed by this PR for 230 Decal Projector:
1. https://unity-ci.cds.internal.unity3d.com/job/10214336 (Win DX11 Standalone XR mono Linear Foundation)
2. https://unity-ci.cds.internal.unity3d.com/job/10214337 (Win DX11 Playmode XR Mono Linear Foundation)
3. https://unity-ci.cds.internal.unity3d.com/job/10214340 (Win DX12 Standalone XR Mono Linear Foundation)
4. https://unity-ci.cds.internal.unity3d.com/job/10214344 (Win Vulkan Standalone mono Linear Foundation)
5. https://unity-ci.cds.internal.unity3d.com/job/10214388 (Win Vulkan Standalone mono Linear Terrain)
6. https://unity-ci.cds.internal.unity3d.com/job/10214390 (Linux Vulkan Standalone mono Linear Terrain)
7. https://unity-ci.cds.internal.unity3d.com/job/10214394 (Android Vulkan Standalone il2cpp Linear Terrain)
8. https://unity-ci.cds.internal.unity3d.com/job/10214460 (Win Vulkan playmode mono Linear Terrain)
9. https://unity-ci.cds.internal.unity3d.com/job/10214461 (Linux Vulkan playmode mono Linear Terrain)

File: com.unity.render-pipelines.universal/Runtime/Decal/DBuffer/DBufferRenderPass.cs [hunk 1]
```diff
@@ -52,72 +48,64 @@ public DBufferRenderPass(Material dBufferClear, DBufferSettings settings, DecalD
...
-                cmd.GetTemporaryRT(Shader.PropertyToID(m_DBufferDepth.name), depthDesc);
+                RenderingUtils.ReAllocateIfNeeded(ref m_DBufferDepth, depthDesc, name: s_DBufferDepthName);
                 depthHandle = m_DBufferDepth;
             }
-            else
-            {
-                depthHandle = m_CameraDepthAttachment;
-            }
 
             ConfigureTarget(dBufferColorHandles, depthHandle);
```
Comment (lukaschod) [inline com.unity.render-pipelines.universal/Runtime/Decal/DBuffer/DBufferRenderPass.cs:110]: I noticed we usually call ConfigureTarget on camera setup, wouldn't be issue calling it outside it?

Comment (sandy-carter-unity) [reply com.unity.render-pipelines.universal/Runtime/Decal/DBuffer/DBufferRenderPass.cs:110]: I'm not sure. So far, I don't think so.
The bigger issues I had was getting targets from universal renderer too early.

**Comment by phi-lira** [comment]
Merged master to rerun tests as there were some failures that were already fixed in master and that was quite hard to compare.

File: com.unity.render-pipelines.universal/Runtime/Decal/DBuffer/DBufferRenderPass.cs (hunk 1 above)
Comment (lukaschod) [reply com.unity.render-pipelines.universal/Runtime/Decal/DBuffer/DBufferRenderPass.cs:110]: K so if it works it is okay. I think I will split it in my https://github.com/Unity-Technologies/Graphics/pull/6442/files.

**Review by lukaschod — APPROVED** [review]

File: com.unity.render-pipelines.universal/Editor/ShaderPreprocessor.cs [hunk 2]
```diff
@@ -940,7 +940,7 @@ private static ShaderFeatures GetSupportedShaderFeatures(UniversalRenderPipeline
                 if (renderer is UniversalRenderer)
                 {
                     UniversalRenderer universalRenderer = (UniversalRenderer)renderer;
-                    if (universalRenderer.renderingMode == RenderingMode.Deferred)
+                    if (universalRenderer.renderingModeRequested == RenderingMode.Deferred)
```
Comment (lukaschod) [inline com.unity.render-pipelines.universal/Editor/ShaderPreprocessor.cs:943]: Could you add info why this needs to be changed?

File: com.unity.render-pipelines.universal/Runtime/UniversalRenderer.cs [hunk 3]
```diff
@@ -198,12 +198,12 @@ public UniversalRenderer(UniversalRendererData data) : base(data)
...
             m_MotionVectorPass = new MotionVectorRenderPass(m_CameraMotionVecMaterial, m_ObjectMotionVecMaterial);
 
-            if (this.renderingMode == RenderingMode.Forward)
+            if (this.renderingModeRequested == RenderingMode.Forward)
             {
                 m_PrimedDepthCopyPass = new CopyDepthPass(RenderPassEvent.AfterRenderingPrePasses, m_CopyDepthMaterial);
             }
 
-            if (this.renderingMode == RenderingMode.Deferred)
+            if (this.renderingModeRequested == RenderingMode.Deferred)
```
Comment (eh-unity) [inline com.unity.render-pipelines.universal/Runtime/UniversalRenderer.cs:206]: Why these use renderingModeRequested? Shouldn't these depend on the renderingModeActual?

**Review by eh-unity — APPROVED** [review]
Please clarify request vs. actual rendering mode usage in the renderer. Otherwise lgtm.

File: com.unity.render-pipelines.universal/Editor/ShaderPreprocessor.cs (hunk 2 above)
Comment (phi-lira) [reply com.unity.render-pipelines.universal/Editor/ShaderPreprocessor.cs:943]: This was to make it more clear to understand. RenderingMode is the rendering mode settings and then there was another RenderingModeActual which was the one that is the actual one used and considered when falling back. The renderinMode was renamed to renderingModeRequested to be more clear this is the requested mode that comes from settings but not necessary the mode that is used.

File: com.unity.render-pipelines.universal/Runtime/UniversalRenderer.cs (hunk 3 above)
Comment (phi-lira) [reply com.unity.render-pipelines.universal/Runtime/UniversalRenderer.cs:206]: @sandy-carter-unity this is a good point. It seems sensible that the actual rendering mode is figured out very early before all of these settings. 

Actually, thinking about it a little bit now, we should expose a property to get the rendering mode that always return the actual one and the requested / setting is private setting, this makes it less confusing.

File: com.unity.render-pipelines.universal/Editor/ShaderPreprocessor.cs (hunk 2 above)
Comment (sandy-carter-unity) [reply com.unity.render-pipelines.universal/Editor/ShaderPreprocessor.cs:943]: Should I add it to the changelog, maybe?

File: com.unity.render-pipelines.universal/Runtime/UniversalRenderer.cs (hunk 3 above)
Comment (sandy-carter-unity) [reply com.unity.render-pipelines.universal/Runtime/UniversalRenderer.cs:206]: Is this what you mean?
```csharp
private RenderingMode renderingModeRequested => m_RenderingMode;
internal RenderingMode renderingModeActual => ...
```

Comment (sandy-carter-unity) [reply com.unity.render-pipelines.universal/Runtime/UniversalRenderer.cs:206]: At that point, we could remove the `renderingModeRequested` property entirely

Comment (phi-lira) [reply com.unity.render-pipelines.universal/Runtime/UniversalRenderer.cs:206]: Yeah, ok I see the `renderingModeRequested` is private now, but that feels useless tbh as you can just have `m_RenderingMode`?

File: com.unity.render-pipelines.universal/Editor/ShaderPreprocessor.cs (hunk 2 above)
Comment (phi-lira) [reply com.unity.render-pipelines.universal/Editor/ShaderPreprocessor.cs:943]: These are internal / private. No reason to mention in changelog.

**Review by phi-lira — APPROVED** [review]

**Comment by phi-lira** [comment]
This PR introduces some new failures in Android, but given that it fixes several other failures I will take it to improve the overall coverage situation.

**Pull request merged**